automation-tests/test/scratchOrg.ts

```diff
@@ -0,0 +1,317 @@
+import {
+  DefaultTreeItem,
+  InputBox,
+  QuickOpenBox,
+  Workbench
+} from 'wdio-vscode-service';
+import {
+  EnvironmentSettings
+} from './environmentSettings';
+import {
+  utilities
+} from './utilities';
+
+const exec = util.promisify(child_process.exec);
+
+export class ScratchOrg {
+  private testSuiteSuffixName: string = undefined;
+  private reuseScratchOrg = false;
+  private projectFolderPath: string = undefined;
+  private prompt: QuickOpenBox | InputBox = undefined;
+  private scratchOrgAliasName: string = undefined;
+
+  public constructor(testSuiteSuffixName: string, reuseScratchOrg: boolean) {
+    this.testSuiteSuffixName = testSuiteSuffixName;
+    this.reuseScratchOrg = reuseScratchOrg;
+
+    // To have all scratch orgs be reused, uncomment the following line:
+    // this.reuseScratchOrg = true;
+  }
+
+  public async setUp(): Promise<void> {
+    await this.setUpTestingEnvironment();
+    await this.createProject();
+    await this.authorizeDevHub();
+    await this.createDefaultScratchOrg();
+  }
+
+  public async tearDown(): Promise<void> {
+    if (this.scratchOrgAliasName && !this.reuseScratchOrg) {
+      // To use VS Code's Terminal view to delete the scratch org, use:
+      // const workbench = await (await browser.getWorkbench()).wait();
+      // await utilities.executeCommand(workbench, `sfdx force:org:delete -u ${this.scratchOrgAliasName} --noprompt`);
+
+      // The Terminal view can be a bit unreliable, so directly call exec() instead:
+      await exec(`sfdx force:org:delete -u ${this.scratchOrgAliasName} --noprompt`);
+    }
+
+    if (this.projectFolderPath) {
+      await utilities.removeFolder(this.projectFolderPath);
+    }
+  }
+
+  public async setUpTestingEnvironment(): Promise<void> {
+    utilities.log('');
+    utilities.log(`${this.testSuiteSuffixName} - Starting setUpTestingEnvironment()...`);
+
+    const tempFolderPath = path.join(__dirname, '..', 'e2e-temp');
+    this.projectFolderPath = path.join(tempFolderPath, this.tempProjectName);
+    utilities.log(`${this.testSuiteSuffixName} - creating project files in ${this.projectFolderPath}`);
+
+    // Clean up the temp folder, just in case there are stale files there.
+    if (fs.existsSync(this.projectFolderPath)) {
+      await utilities.removeFolder(this.projectFolderPath);
+      await utilities.pause(1);
+    }
+
+    // Now create the folders.
+    if (!fs.existsSync(tempFolderPath)) {
+      await utilities.createFolder(tempFolderPath);
+      await utilities.pause(1);
+    }
+
+    await utilities.createFolder(this.projectFolderPath);
+    await utilities.pause(1);
+
+    utilities.log(`${this.testSuiteSuffixName} - ...finished setUpTestingEnvironment()`);
+    utilities.log('');
+  }
+
+  public async createProject(): Promise<void> {
+    utilities.log('');
+    utilities.log(`${this.testSuiteSuffixName} - Starting createProject()...`);
+
+    const workbench = await (await browser.getWorkbench()).wait();
```

Comment:
dumb q: what do we have to `wait()`?  Also is it possible to `await browser.getWorkbench().wait()`?

Reply:
turns out they aren't needed. I went through and removed the calls to `.wait()`